compiler/rustc_typeck/src/check/callee.rs

```diff
@@ -1,10 +1,12 @@
+#![feature(if_let_guard)]
```

Comment:
I believe you need to add it to `compiler/rustc_typeck/src/lib.rs` rather than here.